packages/playground-ui/src/components/assistant-ui/hooks/use-adapters.ts

```diff
@@ -0,0 +1,48 @@
+import { Agent } from '@mastra/core';
+import { useMastraClient } from '@/contexts/mastra-client-context';
+import { useEffect, useState } from 'react';
+import { VoiceAttachmentAdapter } from '../attachments/voice-adapter';
+import {
+  CompositeAttachmentAdapter,
+  SimpleImageAttachmentAdapter,
+  SimpleTextAttachmentAdapter,
+  SpeechSynthesisAdapter,
+  WebSpeechSynthesisAdapter,
+} from '@assistant-ui/react';
+import { PDFAttachmentAdapter } from '../attachments/pdfs-adapter';
+
+export const useAdapters = (agentId: string) => {
+  const [isReady, setIsReady] = useState(false);
+  const [speechAdapter, setSpeechAdapter] = useState<SpeechSynthesisAdapter | undefined>(undefined);
+  const baseClient = useMastraClient();
+
+  useEffect(() => {
```

Comment:
ℹ️ I had to useEffect it to resolve the async `check` function that verifies if we use the DefaultVoice or if we actually have a voice that does something.

The DefaultVoice throws whenever you call a method on it